server/src/handlers/handlers.ts

```diff
@@ -280,6 +283,24 @@ export function handleCodeAction(params: any): Promise<any> {
   })
 }
 
+/**
+ * Handle document formatting requests.
+ *
+ * @param params - The document formatting parameters.
+ * @returns A promise that resolves to an array of text edits.
+ */
+export const handleDocumentFormatting = (params: DocumentFormattingParams): Promise<TextEdit[]> => {
+  const uri = params.textDocument?.uri
+  const options = params.options
+
+  return new Promise<TextEdit[]>(function (resolve) {
+    const job = engine.enqueueJobWithFlattenedParams(jobs.Request.lspFormatting, { uri, options })
+    socket.eventEmitter.once(job.id, ({ result }: socket.FlixResponse) => {
+      resolve((result ?? []) as unknown as TextEdit[])
```

Comment:
What does ?? as unknown do? Is it checking if result is undefined, the returning the empty array?

And then casting (in terms of typescript to TextEdit[]))?